File: app/src/main/java/com/m/livedate/MainActivity.java
package com.m.livedate;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.m.livedate.animation.AnimationActivity;
import com.m.livedate.custom.TouchPullActivity;
import com.m.livedate.custom.ViewActivity;
import com.m.livedate.handler.HandlerActivity;
import com.m.livedate.kotlin.KActivity;
import com.m.livedate.map.MapActivity;
import com.m.livedate.mvp.ui.MVpActivity;
import com.m.livedate.video.MainVideoActivity;
import com.m.livedate.video.VideoActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
    }


    public void goTo(View view) {
        switch (view.getId()) {
            case R.id.mvvm:
                startActivity(new Intent(this,
                        com.m.livedate.mvvm.ui.MainActivity.class));
                break;
            case R.id.mvp:
                startActivity(new Intent(this,
                        MVpActivity.class));
                break;
            case R.id.custom:
                startActivity(new Intent(this,
                        ViewActivity.class));
                break;
            case R.id.kotlin:
                startActivity(new Intent(this,
                        KActivity.class));
                break;
            case R.id.handler:
                startActivity(new Intent(this,
                        HandlerActivity.class));
                break;
            case R.id.animation:
                startActivity(new Intent(this,
                        AnimationActivity.class));
                break;
            case R.id.video:
                startActivity(new Intent(this,
                        MainVideoActivity.class));
                break;
            case R.id.map:
                startActivity(new Intent(this,
                        MapActivity.class));
                break;
            default:
                break;
        }
    }
}

File: mvp_kotlin/src/main/java/com/m/mvp_kotlin/MainActivity.kt
package com.m.mvp_kotlin

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import androidx.datastore.core.DataStore
import androidx.datastore.preferences.core.Preferences
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.createDataStore
import com.m.mvp_kotlin.bean.MBean
import com.m.mvp_kotlin.utils.DataStoreUtils

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val array = IntArray(10);
        val a: Int = 6678956;
        init(a);
        arrayCreate();
    }

    private fun init(vararg v: Int) {

    }

    private fun arrayCreate() {
        var editor = getSharedPreferences("store", Context.MODE_PRIVATE).edit();
        editor.putString("a", "a")
        editor.apply();

        val array1 = Array(10) { i -> i * i }
        val array2 = IntArray(10);
        val array3 = arrayOfNulls<Int>(5)
        val array4 = intArrayOf(1, 2);

        val array5 = String()
        //创建不可变集合，返回值是List
        var list21 = listOf("Java", "Kotlin", null, "Go")
        println(list21)
        println("listOf的返回对象的实际类型：${list21.javaClass}")
        //创建不可变集合，返回值是list，创建不含null的集合对象
        var list2 = listOfNotNull("Java", "Kotlin", null, "Go")
        println(list2)
        println("listOfNotNull的返回对象的实际类型：${list2.javaClass}")
        //创建可变集合，返回值是MutableList
        var mutableList = mutableListOf("Java", "Kotlin", null, "Go")
        println(mutableList)
        println("mutableListOf的返回对象的实际类型：${mutableList.javaClass}")
        //创建ArrayList集合
        var arrayList = arrayListOf("Java", "Kotlin", null, "Go")
        println(arrayList)
        println("arrayListOf的返回对象的实际类型：${arrayList.javaClass}")


        //创建不可变集合，返回值是List
        var list1 = mutableListOf("Java", "Kotlin", null, "Go")
        //在索引2处插入一个新元素
        list1.add(2, "Java")
        println(list1)
        //删除索引1处的元素
        list1.removeAt(1)
        println(list1)
        //将索引1处的元素替换为"Python"
        list1[1] = "Python"
        println(list1)
        //清空List集合的所有元素
        list1.clear()
        println(list1.count())

        val items = setOf(2, 3, 5);
//        items.forEach(fun(value: Int) {
//            print(value);
//        })
        val a: Int = 2;
        val b: Int = 5;
        val max = if (a > b) a else b;

        when (a) {
            1 -> print("x==1")
            2 -> print("x==2")
        }
        val mBean = MBean();
        val name = mBean.name;
        mBean.name = 100;
        mBean.title = "666"
        Log.d("SSSSSSSSSSS", "arrayCreate: " + mBean.title)
        println("SSSSSSSSSSS$mBean.title")
    }

    suspend fun switchView(view: View) {
        when (view.id) {
            R.id.text -> {
                var view = LayoutInflater.from(this).inflate(R.layout.second_layout, null);
                //setContentView(view);
               // DataStoreUtils.writeData(this,"66666")
                //Log.d("SSSSSSSSSSSSSSSS", "switchView: "+DataStoreUtils.readData(this))
            }
        }

    }



}


